File: API/Entities/Plan.cs
namespace API.Entities;

public class Plan : BaseEntity
{
    public string Name { get; set; } = string.Empty;

    public string Slug { get; set; } = string.Empty;

    public string? Description { get; set; }

    public string? ShortDescription { get; set; }

    public int DisplayOrder { get; set; }

    public bool IsFeatured { get; set; }

    public bool IsActive { get; set; } = true;

    public ICollection<PlanLicense> PlanLicenses { get; set; } = new List<PlanLicense>();
}

File: spk/Client/src/models/plan.ts
export interface PlanScopeSummary {
  courseCount: number
  topicCount: number
  questionCount: number
  quizCount: number
  materialCount: number
  estimatedStudyHours: number
}

export interface PlanLicenseSummary {
  id: string
  name: string
  slug: string
  courseCount: number
  topicCount: number
  questionCount: number
  quizCount: number
  materialCount: number
  estimatedStudyHours: number
  hasAccess: boolean
}

export interface Plan {
  id: string
  name: string
  slug: string
  description?: string | null
  shortDescription?: string | null
  displayOrder: number
  isFeatured: boolean
  isActive: boolean
  hasAccess: boolean
  activeLicenseCount: number
  scope: PlanScopeSummary
  licenses: PlanLicenseSummary[]
}
